Detectors/TRD/base/src/LTUParam.cxx

```diff
@@ -0,0 +1,240 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <stdio.h>
+#include "TMath.h"
+#include "TRDBase/LTUParam.h"
+#include "fairlogger/Logger.h"
+
+using namespace o2::trd;
+
+// definition of geometry constants
+std::array<float, 30> LTUParam::mgZrow = {
+  301, 177, 53, -57, -181,
+  301, 177, 53, -57, -181,
+  315, 184, 53, -57, -188,
+  329, 191, 53, -57, -195,
+  343, 198, 53, -57, -202,
+  347, 200, 53, -57, -204
+};
+std::array<float, 6> LTUParam::mgX = { 300.65, 313.25, 325.85, 338.45, 351.05, 363.65 };
+std::array<float, 6> LTUParam::mgTiltingAngle = { -2., 2., -2., 2., -2., 2. };
+int LTUParam::mgDyMax = 63;
+int LTUParam::mgDyMin = -64;
+float LTUParam::mgBinDy = 140e-4;
+std::array<float, 6> LTUParam::mgWidthPad = { 0.635, 0.665, 0.695, 0.725, 0.755, 0.785 };
+std::array<float, 6> LTUParam::mgLengthInnerPadC1 = { 7.5, 7.5, 8.0, 8.5, 9.0, 9.0 };
+std::array<float, 6> LTUParam::mgLengthOuterPadC1 = { 7.5, 7.5, 7.5, 7.5, 7.5, 8.5 };
+float LTUParam::mgLengthInnerPadC0 = 9.0;
+float LTUParam::mgLengthOuterPadC0 = 8.0;
+float LTUParam::mgScalePad = 256. * 32.;
+float LTUParam::mgDriftLength = 3.;
+
+LTUParam::LTUParam() : mMagField(0.),
+                       mOmegaTau(0.),
+                       mPtMin(0.1),
+                       mNtimebins(20 << 5),
+                       mScaleQ0(0),
+                       mScaleQ1(0),
+                       mPidTracklengthCorr(false),
+                       mTiltCorr(false),
+                       mPidGainCorr(false)
+{
+  // default constructor
+}
+
+LTUParam::~LTUParam() = default;
+
+int LTUParam::getDyCorrection(int det, int rob, int mcm) const
+{
+  // calculate the correction of the deflection
+  // i.e. Lorentz angle and tilt correction (if active)
+
+  int layer = det % 6;
+
+  float dyTilt = (mgDriftLength * TMath::Tan(mgTiltingAngle[layer] * TMath::Pi() / 180.) *
+                  getLocalZ(det, rob, mcm) / mgX[layer]);
```

Comment:
I have a few thoughts/suggestions (which depending on the use of this function might be more or less relevant but I would just like to raise awareness):
* prefer using std::tan (math functions from the standard library, no need to go through ROOT, and get automatic guarantee that the right precision version will be chosen)
* actually here the tan values could all be cached since they depend only on certain pre-defined layers)
* The division could be replaced by multiplying with mInvX[layer] or similar

Reply:
@sawenzel good point re lookup table for tan.
Removed the divisions, just waiting for a rebuild, to test.

Reply:
I left one TMath function in Nint as it handles the case of .5 differently from std::round.
Not sure if it makes a material difference, so left it in.